lib/instance-validator.js

```diff
@@ -275,10 +275,17 @@ class InstanceValidator {
       if (typeof validator[validatorType] !== 'function') {
         throw new Error('Invalid validator function: ' + validatorType);
       }
+
       const validatorArgs = this._extractValidatorArgs(test, validatorType, field);
+
       if (!validator[validatorType].apply(validator, [valueString].concat(validatorArgs))) {
-      // extract the error msg
-        throw new Error(test.msg || `Validation ${validatorType} on ${field} failed`);
+        // extract the error msg
+        const e = new Error(test.msg || `Validation ${validatorType} on ${field} failed`);
+
+        e.validatorName = validatorType;
+        e.validatorArgs = validatorArgs;
```

Comment:
ah I see, I though you were referencing just lines 285 & 286 - and that seemed a little overkill.
I actually had pretty much `Object.assign` line before but I think I decided that it was a bit too terse :-) ... I'll put this line in.